### PR TITLE
Fixes #29895 - add pulp_deb plugin

### PR DESCRIPTION
**Pull request by m-bucher**
similar to https://github.com/theforeman/puppet-pulpcore/pull/89 :grinning:

### CODE REVIEW

**Comment by ekohl** [comment]
It's failing on EL8 because the package can't be found:
```
  Error: /Stage[main]/Pulpcore::Plugin::Deb/Pulpcore::Plugin[deb]/Package[python3-pulp-deb]/ensure: change from 'purged' to 'present' failed: Execution of '/usr/bin/dnf -d 0 -e 1 -y install python3-pulp-deb' returned 1: Error: Unable to find a match: python3-pulp-deb
```

Since this is working on EL7 it indicates a packaging issue. I can't find it in koji either so it was never built for it.

Other than that the PR looks good.

**Comment by m-bucher** [comment]
> Since this is working on EL7 it indicates a packaging issue. I can't find it in koji either so it was never built for it.

Yes, although I thought this would have been ensured by the change in `comps/comps-katello-pulpcore-el8.xml` in foreman-packaging (https://github.com/theforeman/foreman-packaging/commit/754a6077cb5af1a7e779b6841fe77e63bf4ab3ee)

Anything I can do to fix this?

**Comment by ekohl** [comment]
It's missing in tito.props. That currently only lists the EL7 tag which means no EL8 package is built.

cc @evgeni for awareness.

**Comment by evgeni** [comment]
We don't whitelist for pulpcire el8, so it's only missing comps, not build.

**Comment by evgeni** [comment]
Turns out it wasn't built because it was not added to koji. Fixed and built.

**Comment by ekohl** [comment]
> Turns out it wasn't built because it was not added to koji. Fixed and built.

With whitelisting we at least had a command that suggested the right commands ;)

**Comment by evgeni** [comment]
> > Turns out it wasn't built because it was not added to koji. Fixed and built.
> 
> With whitelisting we at least had a command that suggested the right commands ;)

Not sure I follow.

**Comment by ekohl** [comment]
I meant that an explicit whitelist had the benefit that we could verify Koji was correctly configured.

**Comment by wbclark** [comment]
@m-bucher could you please rebase when you have a moment? Thanks

**Comment by m-bucher** [comment]
@wbclark done.
Not sure if the test-failures are related.

**Comment by wbclark** [comment]
On a test machine, I'm able to install `python3-pulp-deb` now.

Restarted the CI build and let's see if it passes this time :)

Thanks for your patience @m-bucher

File: spec/classes/plugin_deb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe 'pulpcore::plugin::deb' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+
+      it { is_expected.to compile.with_all_deps }
+      it { is_expected.to contain_pulpcore__plugin('deb') }
```
Comment (wbclark) [inline spec/classes/plugin_deb_spec.rb:9]:
```suggestion
      it { is_expected.to compile.with_all_deps }
      it { is_expected.to contain_pulpcore__plugin('deb') }
      it { is_expected.to contain_package('python3-pulp-deb') }
```

To keep up with the changes in https://github.com/theforeman/puppet-pulpcore/commit/b70670825cc99f1aab28f9cb3192a00ad461c646

@m-bucher could you add this please before I merge ?

**Comment by wbclark** [comment]
Closing and reopening to poke Travis....

**Comment by wbclark** [comment]
@m-bucher GH won't allow me to reopen this without creating a separate PR. Could you do it please so that you are properly credited?

**Comment by m-bucher** [comment]
@wbclark I was also unable to re-open this one, so I created a new one: #112

**Pull request closed without merging**